backend/src/map_model.rs

```diff
@@ -124,8 +139,48 @@ impl Intersection {
         }
     }
 
-    pub fn roads_iter<'a>(&'a self, map: &'a MapModel) -> impl Iterator<Item = &'a Road> {
-        self.roads.iter().map(move |road_id| map.get_r(*road_id))
+    pub fn allowed_movements_from<'a>(
+        &'a self,
+        from_r: RoadID,
+        router_input: &'a impl RouterInput,
+    ) -> impl Iterator<Item = (RoadID, Direction)> + 'a {
+        debug_assert!(
+            self.roads.contains(&from_r),
+            "{from_r:?} is not connected to intersection {self:?}"
+        );
+        let from_road = router_input.get_r(from_r);
+        self.roads.iter().filter_map(move |road_id| {
+            let to_road = router_input.get_r(*road_id);
+            if to_road.id == from_road.id {
```

Comment:
I do not have any counter-examples and will have to think about how to force one, but this rule looks like it prohibits U-turns at a deadend. I don't think a route would ever choose to go down one, unless maybe there's a situation where it's forced to by a TR... ie:

```
       D
       |
  A--*--B
```

If the route starts from B, goes to *, there's a TR preventing B->A, then the route goes to D. It needs to u-turn there to make progress. Then southbound, D->A.

I don't think something this weird is likely to happen in practice, so don't block this PR on it. I've made a note to try and construct this test case later and see if it can actually occur; can file a followup issue if you agree it's possible

Reply:
This was a pre-existing TODO that you just moved around, definitely fine to leave for later

Reply:
I've created https://github.com/a-b-street/ltn/issues/137 to track it. My instinct is to wait and see if it comes up in practice to help determine how important this is to implement.

Reply:
To be clear though, I think you're correct: The proposed routing does not support u-turns.